src/accelerate/utils/bnb.py

```diff
@@ -0,0 +1,390 @@
+import logging
+import os
+from typing import Dict, List, Optional, Union
+
+import torch
+import torch.nn as nn
+
+from accelerate.utils.imports import (
+    is_4bit_bnb_available,
+    is_8bit_bnb_available,
+    is_bnb_available,
+)
+
+from ..big_modeling import dispatch_model, init_empty_weights
+from .dataclasses import BnbQuantizationConfig
+from .modeling import find_tied_parameters, get_balanced_memory, infer_auto_device_map, load_checkpoint_in_model
+
+
+if is_bnb_available():
+    import bitsandbytes as bnb
+
+from copy import deepcopy
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_quantized_model(
+    model: torch.nn.Module,
+    bnb_quantization_config: BnbQuantizationConfig,
+    weights_location: Union[str, os.PathLike] = None,
+    device_map: Optional[Dict[str, Union[int, str, torch.device]]] = None,
+    no_split_module_classes: Optional[List[str]] = None,
+    max_memory: Optional[Dict[Union[int, str], Union[int, str]]] = None,
+    offload_folder: Optional[Union[str, os.PathLike]] = None,
```

Comment:
It's okay with me if support for this is added in a follow-up PR.
Offloading quantized weights in general is probably a bit more work and might require changes in bitsandbytes (last time I tried it was far from easy!)

Reply:
Yeah, I think it will be better to add that in a follow-up PR !